main.py

```diff
@@ -1060,24 +1214,39 @@ def main():
 
             elif packet.is_type(SMSG_TRADE_COMPLETE):
                 commitMessage=""
-                # The sale_tree is only ammended after a complete trade packet.
+                # we get here which tree it should be through UID
+                if trader_state.item.get == 1:
```

Comment:
Tested it on local and this was the lesser of the problems. Actually no problem at all, it didn't occur so I think that or it has to do with this something I just corrected (quite improbable to be honest but it's a program so who knows), or something is returning badly on test, since trader_state.item.get is defined when someone does !add, !buy or !getback (as they check for wrong syntax/uid/item name/etc before setting a new Item() so it could be a weird null item at all or something that surpassed those checkings? As far as I tested here no abnormal syntax passed though them.)